src/app/clusters/pump-configuration-and-control-server/pump-configuration-and-control-server.cpp

```diff
@@ -18,19 +18,321 @@
+            // The ControlMode can be any of the following:
+            // ConstantSpeed, ConstantPressure, ProportionalPressure,
+            // ConstantFlow, ConstantTemperature or Automatic. The actual ControlMode
+            // which would be the EffectiveControlMode is dependant on the actual
+            // physical pump application running "on-top" of this cluster server.
+            Attributes::EffectiveControlMode::Set(endpoint, controlMode);
+            pumpStatus.Clear(PumpStatus::kRemoteFlow);
+            pumpStatus.Clear(PumpStatus::kRemotePressure);
+            pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+            switch (controlMode)
+            {
+            case PumpControlMode::kConstantFlow:
+                pumpStatus.Set(PumpStatus::kRemoteFlow);
+                break;
+            case PumpControlMode::kConstantPressure:
+                pumpStatus.Set(PumpStatus::kRemotePressure);
+                break;
+            case PumpControlMode::kConstantTemperature:
+                pumpStatus.Set(PumpStatus::kRemoteTemperature);
+                break;
+            default:
+                // Intentionally left out
+                break;
+            }
+            break;
+        }
+        // Set the overall effective operation mode to Normal
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kNormal);
+    }
+    break;
+
+        // The pump is controlled by the OperationMode attribute.
+        // Maximum, Minimum or Local
+
+    case PumpOperationMode::kMaximum: {
+        uint8_t maxLevel;
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kMaximum);
+        Attributes::EffectiveControlMode::Set(endpoint, PumpControlMode::kConstantSpeed);
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
+        LevelControl::Attributes::MaxLevel::Get(endpoint, &maxLevel);
+        LevelControl::Attributes::CurrentLevel::Set(endpoint, maxLevel);
+#endif
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+
+    case PumpOperationMode::kMinimum: {
+        uint8_t minLevel;
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kMinimum);
+        Attributes::EffectiveControlMode::Set(endpoint, PumpControlMode::kConstantSpeed);
+#ifdef EMBER_AF_PLUGIN_LEVEL_CONTROL
+        LevelControl::Attributes::MinLevel::Get(endpoint, &minLevel);
+        LevelControl::Attributes::CurrentLevel::Set(endpoint, minLevel);
+#endif
+        pumpStatus.Clear(PumpStatus::kRemoteFlow);
+        pumpStatus.Clear(PumpStatus::kRemotePressure);
+        pumpStatus.Clear(PumpStatus::kRemoteTemperature);
+    }
+    break;
+
+    case PumpOperationMode::kLocal: {
+        Attributes::EffectiveOperationMode::Set(endpoint, PumpOperationMode::kLocal);
+        Attributes::EffectiveControlMode::Set(endpoint, controlMode);
```

Comment:
I'm not sure if this can be set here, given it is local mode I would assume the application code is expected to set the effective control mode to what ever it has as its local config, which might not be what is stored in ControlMode.

Reply:
Ok, thats a question of semantics. It will make good sense, if the ControlMode being set by Matter is "from external entities", and the "internal" ControlMode is set by the application, when the OperationMode is Local. But then the EffectiveControl mode is set by the application. I agree, but perhaps this needs to be clarified i bit?

Reply:
The application could very well also set the OperationMode to Local as well as the external entity (Matter), but the functionality should not collide.

Reply:
Perhaps this could be stated very simple: If the OperationMode is set to kLocal, regardsless of who has set it (external or internal), the application "owns" the EffectiveControlMode.

Reply:
I think the agreement is that if operation mode is local, then it the effective control mode might differ from what is in the ControlMode attribute. But yes, there is two ways into the local mode so we need to figure out how to implement this in a way where we can get the effective control mode set regardless if the operation mode is set as a write from a client or locally on the device